### ISSUE TITLE
Soap API Calling catalogProductAttributeOptionManagementV1->catalogProductAttributeOptionManagementV1GetItems causes PHP error :  PHP Fatal error:  SOAP-ERROR ...

### ISSUE THREAD

**Issue by magento2developer** [magento/magento2#9846]
Note that the other API's I am using seem to be working so far, so this seems specific to catalogProductAttributeOptionManagementV1GetItems 

### Preconditions
Magento 2.1.7
Ubuntu 14.04.5 LTS 
PHP 5.6.30
mysql Ver 14.14
Visual Studio 2017, c#
This is a recent 2.1.5 install that has been upgraded to 2.1.7

### Steps to reproduce
Send Soap Request as below:
POST http://example.com/soap/default?services=catalogProductAttributeOptionManagementV1 HTTP/1.1
Authorization: Bearer 111111111111111111111
Content-Type: application/soap+xml; charset=utf-8; action="catalogProductAttributeOptionManagementV1GetItems"
Host: example.com
Content-Length: 456
Expect: 100-continue
Accept-Encoding: gzip, deflate
Connection: Keep-Alive

<s:Envelope xmlns:s="http://www.w3.org/2003/05/soap-envelope"><s:Body xmlns:xsi="http://www.w3.org/2001/XMLSchema-instance" xmlns:xsd="http://www.w3.org/2001/XMLSchema"><catalogProductAttributeOptionManagementV1GetItemsRequest xmlns="http://example.com/soap/default?services=catalogProductAttributeOptionManagementV1"><attributeCode xmlns="">mount_type</attributeCode></catalogProductAttributeOptionManagementV1GetItemsRequest></s:Body></s:Envelope>

### Expected result
Get a list of options for the drop down attribute.

### Actual result
PHP error logged for Magento server:
PHP Fatal error:  SOAP-ERROR: Encoding: object has no 'label' property in vendor/magento/module-webapi/Model/Soap/Server.php on line 124

Response:
HTTP/1.1 500 Internal Service Error
Date: Sun, 04 Jun 2017 04:41:53 GMT
Server: Apache/2.4.7 (Ubuntu)
Content-Length: 319
X-UA-Compatible: IE=edge
Connection: close
Content-Type: application/soap+xml; charset=utf-8

<?xml version="1.0" encoding="UTF-8"?>
<env:Envelope xmlns:env="http://www.w3.org/2003/05/soap-envelope"><env:Body><env:Fault><env:Code><env:Value>env:Receiver</env:Value></env:Code><env:Reason><env:Text>SOAP-ERROR: Encoding: object has no 'label' property</env:Text></env:Reason></env:Fault></env:Body></env:Envelope>

**Comment by magento2developer** [comment]
Can someone please verify what I have done and perhaps include a patch for the next version.
I suspect that modification i have done will just be overridden next time I upgrade. 

With regard to the fix below, I suspect that an additional type needs to be included on line 160 for the soap handler at vendor/magento/module-webapi/Controller/Soap/Request/Handler.php
I managed to get it working changing line 164
From: $result[$key] = $value; 
To:   $result[$key] =  $this->_dataObjectConverter->convertKeysToCamelCase($this->_dataObjectProcessor->buildOutputDataArray($value, $dataType));

**Comment by TomashKhamlai** [comment]
Thank you for reporting!

I sent SOAP request with body message that you posted here. Got this
```xml
<?xml version="1.0" encoding="UTF-8"?>
<env:Envelope xmlns:env="http://www.w3.org/2003/05/soap-envelope">
    <env:Body>
        <env:Fault>
            <env:Code>
                <env:Value>env:Sender</env:Value>
            </env:Code>
            <env:Reason>
                <env:Text>looks like we got "Body" without function call</env:Text>
            </env:Reason>
        </env:Fault>
    </env:Body>
</env:Envelope>
```

But when I created a new attribute that was created with attributeCode -135, and applied changes that were proposed by you, I was able to send the SOAP-request with message-body given below:
```xml
<?xml version="1.0" encoding="UTF-8"?>
<soap:Envelope xmlns:def="http://example.com/soap/default?services=catalogProductAttributeOptionManagementV1" xmlns:soap="http://www.w3.org/2003/05/soap-envelope">
    <soap:Body>
        <def:catalogProductAttributeOptionManagementV1GetItemsRequest>
	    <attributeCode>135</attributeCode>
        </def:catalogProductAttributeOptionManagementV1GetItemsRequest>
    </soap:Body>
</soap:Envelope>
```

 Internal ticket MAGETWO-53431 is already in our bug-tracking system.

**Comment by TomashKhamlai** [comment]
This bug MAGETWO-53431 is fixed now on Magento 2.2.0. There is no information about backports for earlier Magento releases.

**Comment by magento-engcom-team** [comment]
@magento2developer, thank you for your report.
The issue is already fixed in develop branch, 2.2.0
